src/core/Helpers/Legacy.cpp

```diff
@@ -79,6 +79,7 @@ Drumkit* Legacy::load_drumkit( const QString& dk_path ) {
 				pInstrument->set_filter_active( instrument_node.read_bool( "filterActive", true, false ) );
 				pInstrument->set_filter_cutoff( instrument_node.read_float( "filterCutoff", 1.0f, true, false ) );
 				pInstrument->set_filter_resonance( instrument_node.read_float( "filterResonance", 0.0f, true, false ) );
+				pInstrument->set_pitch_offset( instrument_node.read_float( "pitchOffset", 0.0f, true, false ) );
```

Comment:
So, if I understand what you mean, even if the pitch offset is not set while importing the old drumkit, it is initialised to zero?

Reply:
In Instrument.cpp you already added the initialization of `pitch_offset` to the constructor of `Instrument`. The object `pInstrument` points to does already have its pitch offset set to zero.

`instrument_node.read_float( "pitchOffset", 0.0f, true, false )` will always fail to read the value since it was not present in the legacy drumkits and always returns the default value. It does the same as the initialization in Instrument but now there are two places the default value for `pittch_offset` is defined and they have to be kept in sync. Therefore, I would just drop this line.

Reply:
Perfect!